inst/include/dust/densities.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef DUST_DENSITIES_HPP
+#define DUST_DENSITIES_HPP
+
+#include <cmath>
+
+namespace dust {
+
+template <typename T>
+T maybe_log(T x, bool log) {
```

Comment:
`inline` this I'd say?

Reply:
Possibly inline all the functions here

Reply:
All template functions are inline

Reply:
🤯

Reply:
I feel like I should have known that

Reply:
Seems to depend on the exact definition of "inline", per SO